netfilterqueue/_impl.pyx

```diff
@@ -69,6 +80,8 @@ cdef class Packet:
         cdef nfqnl_msg_packet_hw *hw
         cdef nfqnl_msg_packet_hdr *hdr
 
+        self._nfa = nfa
```

Comment:
This is not safe; the nfq_data pointer only lives as long as the callback, but the Packet object can be retained longer. Not saving nfa to the object is a deliberate choice. You should instead extract the in/out device numbers here (and convert them to names only if requested, since that part is likely to be slower).

Reply:
the way i did this was for performance. this will not punish users who dont care about the in/out interfaces.

also, i dont feel the safety is relevant here. the intention is for the callback to store the in/out values on their side.

once the callback is called, the user can decide what information they want to query and store the returns locally in callback. this makes everything inherently safe because we dont need to rely on dead pointers and subsequent calls dont need to reach into Cython.